classes/class.ilAdobeConnectDfnXMLAPI.php

```diff
@@ -126,99 +127,83 @@ public function externalLogin($user = null, $pass = null, $session = null )
 		$url = $this->getApiUrl(array(
 			'action' 	=> 'lms-user-login',
 			'login' 	=> $user,
-			'session' 	=> $session
+			'session' => $session
 		));
 
-		$context = array(
-			'http' => array('timeout' => 4),
-			'https' => array('timeout' => 4)
-		);
-
-		$ctx = $this->proxy($context);
-		$xml_string = file_get_contents($url, false, $ctx);
-		$xml = simplexml_load_string($xml_string);
-
-		if($xml->status['code'] == 'ok')
+		$xml = $this->sendRequest($url);
+		if($xml instanceof \SimpleXMLElement && $xml->status['code'] == 'ok')
 		{
 			return (string)$xml->cookie;
 		}
 
-		$ilLog->write('AdobeConnect lms-user-login Request: '.$url);
-		$ilLog->write('AdobeConnect lms-user-login failed:  '.$user);
+		$ilLog->error('AdobeConnect lms-user-login Request: '.$url);
+		$ilLog->error('AdobeConnect lms-user-login failed:  '.$user);
 		ilUtil::sendFailure($lng->txt('login_failed'));
 		return false;
 	}
 
-	/**
-	 * @param string $user
-	 * @param string $pass
-	 * @param string $session
-	 * @return bool
-	 */
-	public function login($user, $pass, $session)
-	{
-		global  $DIC;
-		$ilLog = $DIC->logger()->root();
-		$lng = $DIC->language();
-
-		if(isset(self::$loginsession_cache[$session]))
-		{
-			return true;
-		}
-
-		$url = $this->getApiUrl(array(
-			'action' 		=> 'login',
-			'login' 		=> $user,
-			'password' 		=> $pass,
-			'session' 		=> $session
-		));
-
-		$context = array(
-			'http' => array(
-				'timeout' => 4
-			),
-			'https' => array(
-				'timeout' => 4
-			)
-		);
-
-		$ctx = $this->proxy($context);
-		$xml_string = file_get_contents($url, false, $ctx);
-		$xml = simplexml_load_string($xml_string);
-
-		if($xml->status['code'] == 'ok')
-		{
-			self::$loginsession_cache[$session] = true;
-			return true;
-		}
-		else
-		{
-			unset(self::$loginsession_cache[$session]);
-			$ilLog->write('AdobeConnect login Request: '.$url);
-			if($xml)
-			{
-				$ilLog->write('AdobeConnect login Response: ' . $xml->asXML());
-			}
-			$ilLog->write('AdobeConnect login failed: '.$user);
-			ilUtil::sendFailure($lng->txt('login_failed'));
-			return false;
-		}
-	}
-
-	/**
-	 * @param String $session
-	 * @return bool|void
-	 */
-	public function logout($session)
-	{
-	}
+//	/**
+//	 * @param string $user
+//	 * @param string $pass
+//	 * @param string $session
+//	 * @return bool
+//	 */
+//	public function login($user, $pass, $session)
```

Comment:
We should not host code which is wrapped in comments. We have the GIT history for this :smile: .

Reply:
done